distributed/client.py

```diff
@@ -4007,6 +4007,37 @@ async def _get_task_stream(
         else:
             return msgs
 
+    async def _register_scheduler_plugin(self, plugin=None, idempotent=False, **kwargs):
+        if isinstance(plugin, type):
+            plugin = plugin(**kwargs)
+
+        return await self.scheduler.register_scheduler_plugin(
+            plugin=dumps(plugin, protocol=4)
+        )
+
+    def register_scheduler_plugin(self, plugin=None, idempotent=False, **kwargs):
+        """Register a scheduler plugin.
+
+        See https://distributed.readthedocs.io/en/latest/plugins.html#scheduler-plugins
+
+        Parameters
+        ----------
+        plugin : SchedulerPlugin
+            Plugin class or object to pass to the scheduler.
+        idempotent : bool
+            If multiple registrations of the plugin induce no changes.
+        **kwargs : Any
+            Arguments passed to the Plugin class (if Plugin is an
+            instance kwargs are unused).
+
+        """
+        return self.sync(
+            self._register_scheduler_plugin,
+            plugin=plugin,
+            idempotent=idempotent,
```

Comment:
Since this is a new method, and we'll probably end up deprecating `idempotent=` anyways when we go refactor to handle unregistering plugins, I'll suggest we don't support `idempotent=` from the start with `Client.register_scheduler_plugin`

Reply:
Sounds good, removed it.